pkg/agent/client.go

```diff
@@ -314,6 +312,7 @@ func (a *Client) initializeAuthContext(ctx context.Context) (context.Context, er
 // The requests include things like opening a connection to a server,
 // streaming data and close the connection.
 func (a *Client) Serve() {
+	defer a.cs.RemoveClient(a.serverID)
```

Comment:
Again RemoveClient() should be idempotent. So not seeing a problem adding this here.